mirrord/cli/src/container.rs

```diff
@@ -0,0 +1,180 @@
+use std::{
+    ffi::OsStr,
+    io::{BufRead, BufReader, Cursor},
+    net::SocketAddr,
+    path::Path,
+};
+
+use exec::execvp;
+use mirrord_analytics::{AnalyticsError, AnalyticsReporter, CollectAnalytics, Reporter};
+use mirrord_config::LayerConfig;
+use mirrord_progress::{Progress, ProgressTracker};
+use tokio::process::Command;
+
+use crate::{
+    config::{ContainerArgs, ContainerCommand},
+    connection::AGENT_CONNECT_INFO_ENV_KEY,
+    container::command_builder::RuntimeCommandBuilder,
+    error::Result,
+    execution::{MirrordExecution, LINUX_INJECTION_ENV_VAR},
+};
+
+mod command_builder;
+
+static MIRRORD_CONNECT_TCP_ENV_VAR: &str = "MIRRORD_CONNECT_TCP";
+
+fn add_config_path<P>(command: &mut RuntimeCommandBuilder, config_path: P)
+where
+    P: AsRef<OsStr>,
+{
+    let host_path = config_path
+        .as_ref()
+        .to_str()
+        .expect("should convert")
+        .to_owned();
+
+    let config_name = Path::new(&config_path)
+        .file_name()
+        .and_then(|os_str| os_str.to_str())
+        .unwrap_or_default();
+
+    let continer_path = format!("/tmp/{}", config_name);
+
+    command.add_env("MIRRORD_CONFIG_FILE", &continer_path);
+    command.add_volume(host_path, continer_path);
+}
+
+#[tracing::instrument(level = "trace", ret)]
+async fn exec_and_get_first_line(command: &mut Command) -> Result<Option<String>> {
+    let result = command.output().await.expect("TODO: Replace");
+
+    let reader = BufReader::new(Cursor::new(result.stdout));
+
+    reader
+        .lines()
+        .next()
+        .transpose()
+        .map_err(|err| todo!("{err}"))
+}
+
+async fn create_sidecar_intproxy(
+    args: &ContainerArgs,
+    base_command: &RuntimeCommandBuilder,
+    connection_info: Vec<(&str, &str)>,
+) -> Result<(String, SocketAddr)> {
+    let mut sidecar_command = base_command.clone();
+
+    if let Ok(console_addr) = std::env::var("MIRRORD_CONSOLE_ADDR") {
+        sidecar_command.add_env("MIRRORD_CONSOLE_ADDR", console_addr);
+    }
+    sidecar_command.add_env("MIRRORD_INTPROXY_DETACH_IO", "false");
+
+    sidecar_command.add_envs(connection_info);
+
+    let sidecar_container_command = ContainerCommand::Run {
+        runtime_args: vec![
+            "--rm".to_string(),
+            "-d".to_string(),
+            args.cli_image.clone(),
+            "mirrord".to_string(),
+            "intproxy".to_string(),
+        ],
+    };
+
+    let (runtime_binary, sidecar_args) = sidecar_command
+        .with_command(sidecar_container_command)
+        .into_execvp_args();
+
+    // We skip first index of sidecar_args because `RuntimeCommandBuilder::into_execvp_args` adds
+    // the binary as first arg for `execvp`
+    let sidecar_container_id =
+        exec_and_get_first_line(Command::new(&runtime_binary).args(&sidecar_args[1..]))
+            .await?
+            .unwrap();
+
+    let intproxt_socket: SocketAddr =
```

Comment:
Am I seeing double? Could you please add a comment here with that running it with `logs` means?